x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/RequestTask.java

```diff
@@ -38,44 +30,13 @@ class RequestTask implements RejectableTask {
         ActionListener<InferenceServiceResults> listener
     ) {
         this.requestCreator = Objects.requireNonNull(requestCreator);
-        this.listener = getListener(Objects.requireNonNull(listener), timeout, Objects.requireNonNull(threadPool));
+        this.timedListener = new TimedListener<>(timeout, listener, threadPool);
```

Comment:
I moved this into `TimedListener` so we could access it in the new send method of `HttpRequestSender`.